gamemodes/terrortown/entities/weapons/weapon_ttt_binoculars.lua

```diff
@@ -158,6 +170,18 @@ end
 ---
 -- @ignore
 function SWEP:Reload()
+	local make_sound = false
+	if self:GetZoomAmount() ~= 1 or self:GetProgress() > 0 then
+		self:SetZoomLevel(1)
```

Comment:
Why is there zoomAmount and ZoomLevel?

Also normally I'd prefer >= 1 if its handled fluently between a range of 0-1 or directly used as a boolean if only 0 and 1 are used.

Reply:
ZoomLevel is the accessor which does extra logic, ZoomAmount is the actual network variable which stores the most recent value which previously collided with the generic Zoom functions on the weapon base

the not-equals 1 here is to prevent view-punch that occurs when SetZoomLevel is used if the player is already at the smallest zoom amount (no-zoom), and it isn't 0 because it indexes into the zoomlevels list

i will change it to `> 1` though

Reply:
ah I see, thanks